orderer/common/follower/follower_chain.go

```diff
@@ -107,44 +111,52 @@ func NewChain(
 	clusterConsenter consensus.ClusterConsenter,
 	joinBlock *common.Block,
 	options Options,
-	createBlockPullerFunc CreateBlockPuller,
-	chainCreationCallback CreateChain,
+	blockPullerFactory BlockPullerFactory,
+	chainCreator ChainCreator,
 	cryptoProvider bccsp.BCCSP,
 ) (*Chain, error) {
-	// Check the block puller creation function once before we start the follower.
-	puller, errBP := createBlockPullerFunc()
-	if errBP != nil {
-		return nil, errors.Wrap(errBP, "error creating a block puller from join-block")
-	}
-	defer puller.Close()
-
 	options.applyDefaults()
 
 	chain := &Chain{
-		stopChan:              make(chan (struct{})),
-		doneChan:              make(chan (struct{})),
-		cRel:                  types.ClusterRelationFollower,
-		status:                types.StatusOnBoarding,
-		ledgerResources:       ledgerResources,
-		clusterConsenter:      clusterConsenter,
-		joinBlock:             joinBlock,
-		firstHeight:           ledgerResources.Height(),
-		options:               options,
-		logger:                options.Logger.With("channel", ledgerResources.ChannelID()),
-		timeAfter:             options.TimeAfter,
-		createBlockPullerFunc: createBlockPullerFunc,
-		chainCreationCallback: chainCreationCallback,
-		cryptoProvider:        cryptoProvider,
+		stopChan:           make(chan (struct{})),
+		doneChan:           make(chan (struct{})),
+		cRel:               types.ClusterRelationFollower,
+		status:             types.StatusOnBoarding,
+		ledgerResources:    ledgerResources,
+		clusterConsenter:   clusterConsenter,
+		joinBlock:          joinBlock,
+		firstHeight:        ledgerResources.Height(),
+		options:            options,
+		logger:             options.Logger.With("channel", ledgerResources.ChannelID()),
+		timeAfter:          options.TimeAfter,
+		blockPullerFactory: blockPullerFactory,
+		chainCreator:       chainCreator,
+		cryptoProvider:     cryptoProvider,
 	}
 
-	if joinBlock == nil {
-		chain.status = types.StatusActive
+	if clusterConsenter == nil {
```

Comment:
These checks are also redundant to me.  We are the only ones calling this, is there some way we accidentally inject a nil? Wouldn't this be caught in the most basic of integration tests if any of these could be nil?

IMO the only time we should be testing for nil is when we either:
1. Expect the caller to pass in nil as a valid parameter value.
2. Are sanitizing inputs (e.g. reading a protobuf message, user input, etc.)

Reply:
done